src/main/java/org/opensearch/knn/common/KNNConstants.java

```diff
@@ -122,4 +122,6 @@ public class KNNConstants {
     // Please refer this github issue for more details for choosing this value:
     // https://github.com/opensearch-project/k-NN/issues/1049#issuecomment-1694741092
     public static int MAX_DISTANCE_COMPUTATIONS = 2048000;
+    public static int MAX_ID_SELECT_ARRAY = 2000000;
```

Comment:
this value is too high. why we are not building a logic like this:

```
FilterIdsSelectorType getIdSelectorType(const int* filterIds, int filterIdsLength) {
    int maxIdValue = filterIds[filterIdsLength - 1];
    if(filterIdsLength * sizeof(faiss::idx_t) * 8 <= maxIdValue ) {
        return BATCH;
    }
    return BITMAP;
}
```

Reply:
yes you are right. i need reuse this logic, but i miss the code. i will remove MAX_ID_SELECT_ARRAY. as the following code shows:

https://github.com/opensearch-project/k-NN/blob/a48a928eba4b468bf64f6adc1f82f88e38749dbc/src/main/java/org/opensearch/knn/index/query/FilterIdsSelector.java#L83-L89